packages/server/src/api/routes/utils/validators.ts

```diff
@@ -267,13 +275,35 @@ function generateStepSchema(allowStepTypes: string[]) {
     tagline: Joi.string().required(),
     icon: Joi.string().required(),
     params: Joi.object(),
+    inputs: Joi.when("stepId", {
+      is: AutomationActionStepId.BRANCH,
+      then: Joi.object({
+        branches: Joi.array().items(branchSchema).min(1).required(),
+        children: Joi.object()
+          .pattern(Joi.string(), Joi.array().items(Joi.link("#step")))
+          .required(),
+      }).required(),
+      otherwise: Joi.object(),
+    }),
```

Comment:
This is really cool. I had no idea you could do this.

Reply:
I know right, I noticed that Adria had done it elsewhere in the file so I copied him, can't take the credit :)